homeassistant/components/template/trigger.py

```diff
@@ -52,51 +52,51 @@ def template_listener(event, updates):
         if not result_as_boolean(result):
             return
 
-        entity_id = event.data.get("entity_id")
-        from_s = event.data.get("old_state")
-        to_s = event.data.get("new_state")
+        entity_id = event and event.data.get("entity_id")
+        from_s = event and event.data.get("old_state")
+        to_s = event and event.data.get("new_state")
+
+        if entity_id is not None:
+            description = f"{entity_id} via template"
+        else:
+            description = "time change or manual update via template"
+
+        trigger_variables = {
+            "trigger": {
+                "platform": platform_type,
+                "entity_id": entity_id,
+                "from_state": from_s,
+                "to_state": to_s,
+                "for": time_delta,
```

Comment:
I am a bit puzzled by `time_delta` here. Would it work with simply `None`?

This change seems to make the `for` template available as a variable when rendering itself; it seems a bit strange to me but maybe it is intended?

Reply:
Yes, you're right. 

```suggestion
                "for": None,
```

Reply:
There is actually a test for this.  I'll revert the change and split the hash.